src/main/java/Mysql/JAXBMarshaller.java

```diff
@@ -0,0 +1,242 @@
+package Mysql;
+
+import Entities.BaseEntity;
+import jakarta.xml.bind.JAXBContext;
+import jakarta.xml.bind.Unmarshaller;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import java.io.File;
+
+public class JAXBMarshaller {
+
+    public void xmlParser(String XMLFilePath) {
+        try {
+            String[] classes = {"Customer", "Category", "Address", "Supplier", "Order"};
+            File xmlFile = new File(XMLFilePath);
+            DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder builder = factory.newDocumentBuilder();
+            Document document = builder.parse(xmlFile);
+            Element rootElement = document.getDocumentElement();
+
+            for (String a_class : classes) {
+                Class c = Class.forName("Entities." + a_class);
+                JAXBContext customerContext = JAXBContext.newInstance(c);
+                Unmarshaller userUnmarshaller = customerContext.createUnmarshaller();
+
+                 NodeList customerNodes = rootElement.getElementsByTagName(a_class.toLowerCase());
+                for (int i = 0; i < customerNodes.getLength(); i++) {
+                    Node customerNode = customerNodes.item(i);
+
+                    if (customerNode.getNodeType() == Node.ELEMENT_NODE) {
+                        Element customerElement = (Element) customerNode;
+                        BaseEntity baseEntity = (BaseEntity) userUnmarshaller.unmarshal(customerNode);
+                        System.out.println(baseEntity.toString());
+                    }
```

Comment:
Here you are parsing your xml using DOM not using JAXB. You need to parse your xml using JAXB.